Sources/MapboxMaps/Foundation/MapView.swift

```diff
@@ -174,30 +173,94 @@ open class MapView: UIView {
         return mapboxMap.anchor
     }
 
+    private let interfaceOrientationProvider: InterfaceOrientationProvider
+    internal let attributionUrlOpener: AttributionURLOpener
+
     /// Initialize a MapView
     /// - Parameters:
     ///   - frame: frame for the MapView.
     ///   - mapInitOptions: `MapInitOptions`; default uses
     ///    `ResourceOptionsManager.default` to retrieve a shared default resource option, including the access token.
-    public init(frame: CGRect, mapInitOptions: MapInitOptions = MapInitOptions()) {
-        dependencyProvider = MapViewDependencyProvider()
+    @available(iOSApplicationExtension, unavailable)
+    public convenience init(frame: CGRect, mapInitOptions: MapInitOptions = MapInitOptions()) {
```

Comment:
We may need to make this one a designated initializer as well to avoid breaking existing subclasses that use this when calling super.init.

Come to think of it, we could add a MapView subclass for each designated initializer to the test target to avoid regressions in this area.

Reply:
Brilliant idea! I think one `MapView` subclass overriding all designated public initialisers should do the job as well. Is there a reason to have a subclass per each designated initializer?

Reply:
Nope, that works.